File: frontend/src/api/dto/routerProfile.ts
export type RouterProfile = (typeof RouterProfile)[keyof typeof RouterProfile]

export const RouterProfile = {
  BIKE: 'BIKE',
  FASTBIKE: 'FASTBIKE',
  GRAVEL: 'GRAVEL',
  MTB: 'MTB',
  RUN_HIKE: 'RUN_HIKE',
  MOTORCYCLE: 'MOTORCYCLE',
} as const

File: backend/src/main/java/fr/pedalons/dto/router/request/RouterProfile.java
package fr.pedalons.dto.router.request;

import java.util.Map;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum RouterProfile {
  BIKE("bicycle", Map.of("bicycle_type", "hybrid")),
  FASTBIKE("bicycle", Map.of("bicycle_type", "road", "use_roads", 1, "avoid_bad_surfaces", 1)),
  GRAVEL("bicycle", Map.of("bicycle_type", "cross")),
  MTB("bicycle", Map.of("bicycle_type", "mountain")),
  RUN_HIKE("pedestrian", Map.of()),
  MOTORCYCLE("motorcycle", Map.of());

  private final String costing;
  private final Map<String, Object> costingOptions;
}
